docs/sender.rst

```diff
@@ -214,172 +282,227 @@ The transaction is automatically committed when the ``with`` block is exited.
-    
-  *Default: 75000 (HTTP) | 600 (TCP).*
+    buf = Buffer()
+    buf.row(
+        'weather_sensor',
+        symbols={'id': 'toronto1'},
+        columns={'temperature': 23.5, 'humidity': 0.49},
+        at=TimestampNanos.now())
 
-* ``auto_flush_bytes`` - ``int > 0`` | ``'off'``: The number of bytes that will
-  trigger a flush. Set to ``'off'`` to disable.
-        
-  *Default: ``'off'``.*
+    conf1 = 'http::addr=db1.host.com:9000;'
+    conf2 = 'http::addr=db2.host.com:9000;'
+    with Sender.from_conf(conf1) as sender1, Sender.from_conf(conf2) as sender2:
+        sender1.flush(buf1, clear=False)
+        sender2.flush(buf2, clear=False)
 
-* ``auto_flush_interval`` - ``int > 0`` | ``'off'``: The time in milliseconds
-  that will trigger a flush. Set to ``'off'`` to disable.
-    
-  *Default: 1000 (millis).*
+    buf.clear()
 
-.. _configuring-sender-buffer:
+This uses the ``clear=False`` parameter which otherwise defaults to ``True``.
 
-Buffer
-------
+Threading Considerations
+------------------------
 
-* ``init_buf_size`` - ``int > 0``: Initial buffer capacity.
-    
-  *Default: 65536 (64KiB).*
+Neither buffer API nor the sender object are thread-safe, but can be shared
+between threads if you take care of exclusive access (such as using a lock)
+yourself.
 
-* ``max_buf_size`` - ``int > 0``: Maximum flushable buffer capacity.
-    
-  *Default: 104857600 (100MiB).*
+Independent buffers also allows you to prepare separate buffers in different
+threads and then send them later through a single exclusively locked sender.
 
-* ``max_name_len`` - ``int > 0``: Maximum length of a table or column name.
+Alternatively you can also create multiple senders, one per thread.
 
-  *Default: 127.*
+Notice that the ``questdb`` python module is mostly implemented in Cython and
+Rust and is designed to release the GIL as much as possible, so you can expect
+good performance in multi-threaded scenarios.
 
-.. _configuring-sender-request:
+As an example, appending a dataframe to a buffer releases the GIL (unless any
+of the columns reference python objects).
 
-HTTP Request
-------------
+All network activity also fully releases the GIL.
 
-The following parameters control the HTTP request behavior.
+.. _sender_http_performance:
 
-* ``retry_timeout`` - ``int > 0``: The time in milliseconds to continue retrying
-  after a failed HTTP request. The interval between retries is an exponential
-  backoff starting at 10ms and doubling after each failed attempt up to a
-  maximum of 1 second.
-    
-  *Default: 10000 (10 seconds).*
+Optimising HTTP Performance
+---------------------------
 
-* ``request_timeout`` - ``int > 0``: The time in milliseconds to wait for a
-  response from the server. This is in addition to the calculation derived from
-  the ``request_min_throughput`` parameter.
-    
-  *Default: 10000 (10 seconds).*
+The sender's network communication is implemented in native Rust and thus does
```

Comment:
Similar to above, perhaps drop this extra context if it is not immediately relevant.

Reply:
Ditto.